torch/distributed/fsdp/fully_sharded_data_parallel.py

```diff
@@ -760,7 +762,8 @@ def _lazy_init(self) -> None:
         self._is_root = True
         self._assert_state(TrainingState.IDLE)
         _init_streams(self)
-        self._cast_buffers(recurse=True)
```

Comment:
do we still need the code `_cast_buffers` implementation, or can it also call into the new ones?

Reply:
We still need the one in `FullyShardedDataParallel`. I did not remove it from state-dict-related methods.

Reply:
I will try to retire the old one in follow-ups.